strawberryfields/backends/states.py

```diff
@@ -914,6 +914,25 @@ def number_expectation(self, modes):
             ps = np.tensordot(np.diag(values), ps, axes=((0, 1), (0, 1)))
         return float(ps)
 
+    def number_expectation(self, modes):
+        """Calculates the expectation value of a product of number operators acting on given modes"""
+        cutoff = self._cutoff
+        values = np.arange(cutoff)
+        return self.diagonal_expectation(modes, values)
+
+    def parity_expectation(self, modes):
+        """Calculates the expectation value of a product of parity operators acting on given modes"""
+        cutoff = self._cutoff
+        values = (-1) ** np.arange(cutoff)
+        return self.diagonal_expectation(modes, values)
+
+    def quadratic_expectation(self, modes, a, b, c):
```

Comment:
I am not sure this function has a very strong use case.

Reply:
Yeah no problem, I'll remove it